lib/cloudevents/format/v_0_1/decoder/json.ex

```diff
@@ -0,0 +1,126 @@
+defmodule Cloudevents.Format.V_0_1.Decoder.JSON do
+  @moduledoc false
+  @behaviour Cloudevents.Format.Decoder.JSON
+
+  alias Cloudevents.Format.Decoder.DecodeError
+  alias Cloudevents.Format.ParseError
+  alias Cloudevents.Format.V_0_1.Event
+
+  @doc """
+  Turns a JSON string into a Cloudevent 0.2 struct.
+
+  ## Examples
+
+  ### Successful case
+
+      iex> json = ~S({
+      ...>   "specversion": "0.2",
+      ...>   "type": "com.github.pull.create",
+      ...>   "source": "https://github.com/cloudevents/spec/pull",
+      ...>   "id": "A234-1234-1234",
+      ...>   "time": "2018-04-05T17:31:00Z",
+      ...>   "comexampleextension1": "value",
+      ...>   "comexampleothervalue": 5,
+      ...>   "contenttype": "text/xml",
+      ...>   "data": "<much wow=\\"xml\\"/>"
+      ...> })
+      iex> {:ok, event} = Cloudevents.Format.V_0_2.Decoder.JSON.decode(json)
+      iex> with %Cloudevents.Format.V_0_2.Event{
+      ...>   type: "com.github.pull.create",
+      ...>   source: "https://github.com/cloudevents/spec/pull",
+      ...>   id: "A234-1234-1234",
+      ...>   time: "2018-04-05T17:31:00Z",
+      ...>   extensions: %{
+      ...>     "comexampleextension1" => "value",
+      ...>     "comexampleothervalue" => 5
+      ...>   },
+      ...>   contenttype: "text/xml",
+      ...>   data: ~S(<much wow="xml"/>)
+      ...> } <- event, do: :passed
+      :passed
+
+  ### Not a JSON at all
+
+      iex> not_a_json = "..."
+      iex> Cloudevents.Format.V_0_2.Decoder.JSON.decode(not_a_json)
+      {:error, %Cloudevents.Format.Decoder.DecodeError{
+                cause: %Jason.DecodeError{data: "...", position: 0, token: nil}}}
+
+  ### Missing required fields
+
+      iex> json = ~S({
+      ...>   "specversion": "0.2",
+      ...>   "type": "com.github.pull.create"
+      ...> })
+      iex> Cloudevents.Format.V_0_2.Decoder.JSON.decode(json)
+      {:error, %Cloudevents.Format.Decoder.DecodeError{
+                cause: %Cloudevents.Format.ParseError{message: "missing source"}}}
+
+  ### Invalid extension attribute name
+
+      iex> json = ~S({
+      ...>   "specversion": "0.2",
+      ...>   "type": "com.github.pull.create",
+      ...>   "source": "https://github.com/cloudevents/spec/pull",
+      ...>   "id": "A234-1234-1234",
+      ...>   "an extension attribute that contains spaces": "is not allowed"
+      ...> })
+      iex> Cloudevents.Format.V_0_2.Decoder.JSON.decode(json)
+      {:error, %Cloudevents.Format.Decoder.DecodeError{
+                cause: %Cloudevents.Format.ParseError{message: "invalid extension attributes: \\"an extension attribute that contains spaces\\""}}}
+
+  """
+  def decode(json) do
+    with {:ok, orig_map} <- Jason.decode(json),
+         preprocessed_map = handle_data_base64(orig_map),
+         {:ok, event} <- Event.from_map(preprocessed_map) do
+      # `data_base64` may only be used for binary data, so we only try to interpret
```

Comment:
I think in 0.1 base64-encoded data isn't even supported, or is it?

Reply:
Seems like it's not.